src/gdbserver/semihosting.c

```diff
@@ -338,20 +338,20 @@ int do_semihosting (stlink_t *sl, uint32_t r0, uint32_t r1, uint32_t *ret) {
         DLOG("Semihosting: read(%d, target_addr:0x%08x, %zu)\n", fd,
              buffer_address, buffer_len);
 
-        read_result = (uint32_t)read(fd, buffer, buffer_len);
+        read_result = read(fd, buffer, buffer_len);
         saved_errno = errno;
 
-        if (*ret == (uint32_t)-1) {
+        if (read_result == (uint32_t)-1) {
```

Comment:
comparison between signed and unsigned integer expressions [-Werror=sign-compare]

Reply:
You are right, read_result is a ssize_t now (which is a signed integer).  I'll fix this too since it's my mistake.